libqtile/manager.py

```diff
@@ -1519,6 +1520,33 @@ def cmd_qtile_info(self):
 
     def cmd_shutdown(self):
         """Quit Qtile"""
+        pids = [win.window.get_net_wm_pid() for _, win in self.windowMap.items()]
+        pids = list(filter(lambda x: x is not None, pids))
+
+        # Give the windows a chance to shut down nicely.
+        for pid in pids:
+            try:
+                os.kill(pid, signal.SIGTERM)
+            except ProcessLookupError:
+                # might have died recently
+                pass
+
+        def still_alive(pid):
+            # most pids will not be children, so we can't use wait()
+            try:
+                os.kill(pid, 0)
+                return True
+            except ProcessLookupError:
+                return False
+
+        # give everyone a little time to exit and write their state. but don't
+        # sleep forever (2s).
+        for _ in range(4):
+            pids = list(filter(still_alive, pids))
+            if len(pids) == 0:
+                break
+            time.sleep(0.5)
```

Comment:
0.5 seconds may be a little extreme, probably a time.sleep(0.1) with 10 loop iterations would be better. I can change it depending on what people think :)

Reply:
I think I'd rather do 0.1 also.

Reply:
Don't we want to use `asyncio.sleep` rather than `time.sleep`?